### PR TITLE
Base folder structure with helm and NFS prov/deprov functionality

### PR DESCRIPTION
**Pull request by nikhilbarge**
This is PR for 
1. Base structure for CSI project
2. Helm Deployment
3. NFS provisioning and deprovisioning functionality

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
+FROM centos:7.4.1708
+# RUN yum install -y libaio
+# RUN yum install -y libuuid
+# RUN yum install -y numactl
+# RUN yum install -y xfsprogs
+# RUN yum install -y e4fsprogs
+RUN yum -y install iscsi-initiator-utils e2fsprogs xfsprogs && yum clean all
```
Comment (gadekarnitesh) [inline Dockerfile:7]: remove the commented code

File: api/apiclient_mock.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+}
+
+func (m *MockApiClient) GetWithQueryString(ctx context.Context, url, queryString string, expectedResp interface{}) (interface{}, error) {
+	args := m.Called()
+	resp, _ := args.Get(0).(interface{})
+	err, _ := args.Get(1).(error)
+	return resp, err
+}
+
+// func (m *MockApiService) CreateVolume(volume *VolumeParam, storagePoolName string) (*VolumeResp, error) {
```
Comment (gadekarnitesh) [inline api/apiclient_mock.go:53]: remove unused methods

File: api/apiclient_test.go [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+	serviceMock *MockApiService
+}
+
+func TestServiceTestSuite(t *testing.T) {
+	suite.Run(t, new(ApiTestSuite))
+}
+
+func (suite *ApiTestSuite) Test_CreateVolumeGetStoragePoolIDByName_Fail() {
+	// Test the storage pool name not available
+	expectedError := errors.New("volume with given name not found")
```
Comment (gadekarnitesh) [inline api/apiclient_test.go:28]: remove commented code from this file

File: api/apiclient.go [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+
+func (c *ClientService) NewClient() (*ClientService, error) {
+	restclient, err := client.NewRestClient()
+	if err != nil {
+		return c, err
+	}
+	c.api = restclient
+	return c, nil
+}
+
```
Comment (gadekarnitesh) [inline api/apiclient.go:59]: add comment on public methods

File: api/nfsapiclient.go [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+	"fmt"
+	"net/http"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/common/log"
+)
+
+func (c *ClientService) OneTimeValidation(poolname string, networkspace string) (list string, err error) {
```
Comment (gadekarnitesh) [inline api/nfsapiclient.go:14]: add comment on public methods

File: helm/infinibox-csi-driver/templates/storageclass-fc.yaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+apiVersion: storage.k8s.io/v1
```
Comment (gadekarnitesh) [inline helm/infinibox-csi-driver/templates/storageclass-fc.yaml:1]: As per discussion, we need to remove create StorageClass ymal from helm deployment

**Comment by talimahar** [comment]
All,

The branch name and commit is wrong 

**Branch name:** INFINIBOX-CSIC-16-packagestructure
It should be withOUT the INFINIBOX prefix
=> CSIC-16-packagestructure


**Commit:** "Base folder structure with helm and NFS prov/deprov functionality"

First, 
Commits should include **JIRA ticket** as well
e.g: "CSCI-xx: Base folder structure with helm and NFS prov/deprov functionality"
Second,
The "correct" jira ticket should be the one that refers the most to the actual developing.

According to Jira
CSCI-16 is an **EPIC** which includes 2 tasks (I changed them to Task instead "New Feature"):
CSIC-12 (Helm Chart - Development and Testing of CSI Packaging)
CSIC-17 (Operator - Development and Testing of CSI Packaging)

As the pushed commit refer to Helm

The correct commit header should be:
"CSCI-**12**: Base folder structure with helm and NFS prov/deprov functionality"

Please fix it.

**Pull request merged**